app/helpers/needs_helper.rb

```diff
@@ -1,14 +1,22 @@
 module NeedsHelper
   def populate_needs
+    rejected = []
+    approved = []
     session[:cart].each do |id, qty|
-      need = NeedType.find(id)
-      current_user.needs.create(name: pluralize(qty, need.name),
-                                description: need.description,
-                                cost: need.cost * qty,
-                                image_url: need.image_url,
-                                slug: need.slug,
-                                category: need.category
-                                )
+      need_type = NeedType.find(id)
+      if current_user.needs.exists?(slug: need_type.slug)
+        rejected << need_type.name
+      else
+        current_user.needs.create(name: pluralize(qty, need_type.name),
+                                  description: need_type.description,
+                                  cost: need_type.cost * qty,
+                                  image_url: need_type.image_url,
+                                  slug: need_type.slug,
+                                  category: need_type.category
+                                  )
+        approved << need_type.name
```

Comment:
If the need isn't present, it is added to the store and the name is added to a different array that's passed to a different flash showing what _was_ allowed into the store.